internal/webhooks/cluster.go

```diff
@@ -362,3 +365,84 @@ func (webhook *Cluster) getClusterClassForCluster(ctx context.Context, cluster *
 	}
 	return clusterClass, nil
 }
+
+func validateMachineHealthCheckTopologies(cluster *clusterv1.Cluster, clusterClass *clusterv1.ClusterClass) field.ErrorList {
+	var allErrs field.ErrorList
+
+	// Validation ControlPlane MachineHealthCheck if defined.
+	if cluster.Spec.Topology.ControlPlane.MachineHealthCheck != nil && !cluster.Spec.Topology.ControlPlane.MachineHealthCheck.MachineHealthCheckClass.IsZero() {
+		// Ensure ControlPlane does not define a MachineHealthCheck if it does not define MachineInfrastructure.
+		if clusterClass.Spec.ControlPlane.MachineInfrastructure == nil {
+			allErrs = append(allErrs, field.Forbidden(
+				field.NewPath("spec", "topology", "controlPlane", "machineHealthCheck", "spec"),
+				"can be set only if spec.controlPlane.machineInfrastructure is set in ClusterClass",
+			))
+		}
+		// Ensure ControlPlane MachineHealthCheck defines UnhealthyConditions.
+		if len(cluster.Spec.Topology.ControlPlane.MachineHealthCheck.MachineHealthCheckClass.UnhealthyConditions) == 0 {
+			allErrs = append(allErrs, field.Forbidden(
+				field.NewPath("spec", "topology", "controlPlane", "machineHealthCheck", "spec", "unhealthyConditions"),
+				"must have at least one value",
+			))
+		}
+	}
+
+	// If MachineHealthCheck is explicitly enabled then make sure that a MachineHealthCheck definition is
+	// available either in the Cluster topology or in the ClusterClass.
+	// (One of these definitions will be used in the controller to create the MachineHealthCheck)
+	if cluster.Spec.Topology.ControlPlane.MachineHealthCheck != nil &&
+		cluster.Spec.Topology.ControlPlane.MachineHealthCheck.Enable != nil &&
+		*cluster.Spec.Topology.ControlPlane.MachineHealthCheck.Enable &&
+		cluster.Spec.Topology.ControlPlane.MachineHealthCheck.MachineHealthCheckClass.IsZero() &&
+		clusterClass.Spec.ControlPlane.MachineHealthCheck == nil {
+		allErrs = append(allErrs, field.Forbidden(
+			field.NewPath("spec", "topology", "controlPlane", "machineHealthCheck", "enable"),
+			fmt.Sprintf("cannot be set to %v as MachineHealthCheck definition is not available in the Cluster topology or the ClusterClass", *cluster.Spec.Topology.ControlPlane.MachineHealthCheck.Enable),
```

Comment:
```suggestion
			fmt.Sprintf("cannot be set to %t as MachineHealthCheck definition is not available in the Cluster topology or the ClusterClass", *cluster.Spec.Topology.ControlPlane.MachineHealthCheck.Enable),
```
let's use %t for bool, please check if there are other occurences